spec/features/work_packages/table/queries/hierarchy_cf_filter_spec.rb

```diff
@@ -0,0 +1,96 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+
+RSpec.describe "Work package filtering by user custom field", :js do
+  let(:project) { create(:project) }
+  let(:type) { project.types.first }
+  let(:wp_table) { Pages::WorkPackagesTable.new(project) }
+  let(:filters) { Components::WorkPackages::Filters.new }
+  let(:hierarchy_root) { create(:hierarchy_item) }
+  let!(:hierarchy_cf) do
+    create(:hierarchy_wp_custom_field, hierarchy_root:).tap do |cf|
+      type.custom_fields << cf
+      project.work_package_custom_fields << cf
+    end
+  end
+  let(:service) { CustomFields::Hierarchy::HierarchicalItemService.new }
+  let!(:luke) { service.insert_item(parent: hierarchy_root, label: "luke").value! }
+  let!(:leia) { service.insert_item(parent: hierarchy_root, label: "leia").value! }
+
+  let!(:wp_luke) do
+    create(:work_package, project:, subject: "Luke's wp").tap do |wp|
+      wp.custom_field_values = { hierarchy_cf.id => luke.id }
+      wp.save!
+    end
+  end
+  let!(:wp_leia) do
+    create(:work_package, project:, subject: "Leia's wp").tap do |wp|
+      wp.custom_field_values = { hierarchy_cf.id => leia.id }
+      wp.save!
+    end
+  end
+
+  let(:admin) { create(:admin) }
+
+  current_user { admin }
+
+  describe "filters" do
+    before do
+      wp_table.visit!
+      wp_table.expect_work_package_listed(wp_luke, wp_leia)
+    end
+
+    context "when any" do
+      it "shows the work package matching the hierarchy cf filter" do
+        filters.open
+
+        # Filtering by hierarchy (=)
+
+        filters.add_filter_by(hierarchy_cf.name, "is", [luke.label], hierarchy_cf.attribute_name(:camel_case))
```

Comment:
❓ ooc: how much effort is it to have a clone of the `is` filter operator which is called `is any`, being more precise in what it is doing?

question opts for maybe suggesting within our current epic scope a renaming (in code) from `is` to `is any`, to easily distinguish between `is any` and `is all` in future.

Reply:
We initially had implemented a custom operator for hierarchies, but after some investigation it turned out the behaviour would be exactly the same as the [Equals operator](https://github.com/opf/openproject/blob/dev/app/models/queries/operators/equals.rb).

And after looking into other operators, I found that there is already a [EqualsAll operator](https://github.com/opf/openproject/blob/dev/app/models/queries/operators/equals_all.rb).

Which leads me to think that using the default operators here is better because it is more aligned with the rest of the application. Meaning I don't think it is better to use explicit `is any` and `is all`.